tests/data_group_test.py

```diff
@@ -1,8 +1,211 @@
 # SPDX-License-Identifier: BSD-3-Clause
 # Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+import numpy as np
+import pytest
+
 import scipp as sc
 
 
+def test_create_from_kwargs():
+    dg = sc.DataGroup(a=4, b=6)
+    assert tuple(dg.keys()) == ('a', 'b')
+
+
+def test_create_from_dict_works_with_mixed_types_and_non_scipp_objects():
+    items = {
+        'a': 1,
+        'b': sc.scalar(2),
+        'c': sc.DataArray(sc.scalar(3)),
+        'd': np.arange(4)
+    }
+    dg = sc.DataGroup(items)
+    assert tuple(dg.keys()) == ('a', 'b', 'c', 'd')
+
+
+def test_init_raises_when_keys_are_not_strings():
+    d = {1: 0}
+    with pytest.raises(ValueError):
+        sc.DataGroup(d)
+
+
+def test_len():
+    assert len(sc.DataGroup()) == 0
+    assert len(sc.DataGroup({'a': 0})) == 1
+    assert len(sc.DataGroup({'a': 0, 'b': 1})) == 2
+
+
+def test_iter_empty():
+    dg = sc.DataGroup()
+    it = iter(dg)
+    with pytest.raises(StopIteration):
+        next(it)
+
+
+def test_iter_nonempty():
+    dg = sc.DataGroup({'a': sc.scalar(1), 'b': sc.scalar(2)})
+    it = iter(dg)
+    assert next(it) == 'a'
+    assert next(it) == 'b'
+    with pytest.raises(StopIteration):
+        next(it)
+
+
+def test_getitem_str_key():
+    dg = sc.DataGroup({'a': sc.scalar(1), 'b': sc.scalar(2)})
+    assert sc.identical(dg['a'], sc.scalar(1))
+    assert sc.identical(dg['b'], sc.scalar(2))
+
+
+def test_setitem_str_key_adds_new():
+    dg = sc.DataGroup()
+    dg['a'] = sc.scalar(1)
+    assert sc.identical(dg['a'], sc.scalar(1))
+
+
+def test_setitem_str_key_replace():
+    dg = sc.DataGroup({'a': sc.scalar(1)})
+    dg['a'] = sc.scalar(2)
+    assert sc.identical(dg['a'], sc.scalar(2))
+
+
+def test_delitem_removes_item():
+    dg = sc.DataGroup({'a': sc.scalar(1)})
+    del dg['a']
+    assert 'a' not in dg
+
+
+def test_delitem_raises_KeyError_when_not_found():
+    dg = sc.DataGroup({'a': sc.scalar(1)})
+    with pytest.raises(KeyError):
+        del dg['b']
+
+
+def test_dims_with_scipp_objects_combines_dims_in_insertion_order():
+    assert sc.DataGroup({'a': sc.scalar(1)}).dims == ()
+    assert sc.DataGroup({'a': sc.ones(dims=('x', ), shape=(2, ))}).dims == ('x', )
+    assert sc.DataGroup({
+        'a': sc.ones(dims=('x', ), shape=(2, )),
+        'b': sc.ones(dims=('y', ), shape=(2, ))
+    }).dims == ('x', 'y')
+    assert sc.DataGroup({
+        'b': sc.ones(dims=('y', ), shape=(2, )),
+        'a': sc.ones(dims=('x', ), shape=(2, ))
+    }).dims == ('y', 'x')
+    # In this case there would be a better order, but in general there is not, so at
```

Comment:
What would the better order be? Not sure the comment is so useful?

Reply:
`['x', 'y', 'z']`, since it avoids reordering anythin.